public/app/features/alerting/unified/components/rule-editor/QueryRows.tsx

```diff
@@ -278,6 +288,20 @@ function copyModel(item: AlertQuery, uid: string): Omit<AlertQuery, 'datasource'
   };
 }
 
+function newModel(item: AlertQuery, ds: DataSourceApi): Omit<AlertQuery, 'datasource'> {
+  return {
+    refId: item.refId,
+    relativeTimeRange: item.relativeTimeRange,
+    queryType: '',
+    datasourceUid: ds.uid,
+    model: {
+      ...ds.getDefaultQuery?.(CoreApp.UnifiedAlerting),
+      refId: item.refId,
+      hide: false,
```

Comment:
Not too familiar with alerting, but you may also want to set data source ref in the model. See how it's done in [panel edit](https://github.com/grafana/grafana/blob/main/public/app/features/query/components/QueryGroup.tsx#L236) for example.

Reply:
iirc we grab the datasource settings from the `datasourceUid` property from the root (not from the model) and pass that in to the query editor, so no need to have it in the model.